app/rpc/rpchandlers/get_mempool_entries_by_addresses.go

```diff
@@ -0,0 +1,88 @@
+package rpchandlers
+
+import (
+	"github.com/kaspanet/kaspad/app/appmessage"
+	"github.com/kaspanet/kaspad/app/rpc/rpccontext"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/txscript"
+	"github.com/kaspanet/kaspad/infrastructure/network/netadapter/router"
+	"github.com/kaspanet/kaspad/util"
+)
+
+// HandleGetMempoolEntriesByAddresses handles the respectively named RPC command
+func HandleGetMempoolEntriesByAddresses(context *rpccontext.Context, _ *router.Router, request appmessage.Message) (appmessage.Message, error) {
+
+	transactions := context.Domain.MiningManager().AllTransactions()
+	getMempoolEntriesByAddressesRequest := request.(*appmessage.GetMempoolEntriesByAddressesRequestMessage)
+	mempoolEntriesByAddresses := make([]*appmessage.MempoolEntryByAddress, 0)
+
+	for _, addressString := range getMempoolEntriesByAddressesRequest.Addresses {
+
+		_, err := util.DecodeAddress(addressString, context.Config.ActiveNetParams.Prefix)
+		if err != nil {
+			errorMessage := &appmessage.GetUTXOsByAddressesResponseMessage{}
+			errorMessage.Error = appmessage.RPCErrorf("Could not decode address '%s': %s", addressString, err)
+			return errorMessage, nil
+		}
+
+		sending := make([]*appmessage.MempoolEntry, 0)
+		receiving := make([]*appmessage.MempoolEntry, 0)
+
+		for _, transaction := range transactions {
+
+			for _, input := range transaction.Inputs {
+				_, transactionSendingAddress, err := txscript.ExtractScriptPubKeyAddress(
+					input.UTXOEntry.ScriptPublicKey(),
+					context.Config.ActiveNetParams)
+				if err != nil {
+					return nil, err
+				}
+				if addressString == transactionSendingAddress.String() {
+					rpcTransaction := appmessage.DomainTransactionToRPCTransaction(transaction)
+					sending = append(
+						sending,
+						&appmessage.MempoolEntry{
+							Fee:         transaction.Fee,
+							Transaction: rpcTransaction,
+						},
+					)
+					break //one input is enough
+				}
+			}
+
+			for _, output := range transaction.Outputs {
+				_, transactionReceivingAddress, err := txscript.ExtractScriptPubKeyAddress(
+					output.ScriptPublicKey,
+					context.Config.ActiveNetParams,
+				)
+				if err != nil {
+					return nil, err
+				}
+				if addressString == transactionReceivingAddress.String() {
+					rpcTransaction := appmessage.DomainTransactionToRPCTransaction(transaction)
+					receiving = append(
+						receiving,
+						&appmessage.MempoolEntry{
+							Fee:         transaction.Fee,
+							Transaction: rpcTransaction,
+						},
+					)
+					break //one output is enough
+				}
+			}
+
+			if len(sending) > 0 || len(receiving) > 0 {
```

Comment:
I suppose this will result in each address having a MempoolEntryByAddress, even if sending / receving is empty. I would like to imitate the behaviour of GetUtxosByAddressess, to have some sort of standard. which i think leaves out all addresses that don't have utxo's. but i will look into it.

Reply:
I added a comment to reflect this discussion. Personally, i think all ByAddresses calls should follow the same convention / standard regarding addresses for which the call cannot find an "entry", in this case it would require the if statement. But, if you insist I will happily oblige. Just wish to raise this point as a counter argument, before potentially changing this.